ballerina/subscription_visitor.bal

```diff
@@ -88,8 +88,8 @@ class SubscriptionVisitor {
         if selection is parser:FragmentNode {
             ErrorDetail errorDetail = getErrorDetailRecord(message, selection.getLocation());
             self.errors.push(errorDetail);
-        } else {
-            ErrorDetail errorDetail = getErrorDetailRecord(message, (<parser:FieldNode>selection).getLocation());
+        } else if selection is parser:FieldNode {
+            ErrorDetail errorDetail = getErrorDetailRecord(message, selection.getLocation());
             self.errors.push(errorDetail);
         }
```

Comment:
Missing `else` block?

Reply:
If the `else`  block is used instead of `else if` , we have to use casting in the next line, to get the location from the selection node. Since the `Selection` node can only be either `FragmentNode` or `FieldNode` , it is not necessary to use `else` block here.

Reply:
Here we need to use the `else` block after that lang change. right? @ThisaruGuruge

Reply:
Since it does not return a value, I think we don't need it. @Nuvindu Let's check whether it complains of any compilation errors here.

Reply:
@ThisaruGuruge I checked with the latest version. No any compilation error occurred here.